### ISSUE TITLE
Error message: "Text must be less than or equal to 140 characters" when it is not over

### ISSUE THREAD

**Issue by tony-mcgee** [bear/python-twitter#457]
Hello,

I use this great library to post data from a weather station and camera. Every hour, the current weather data plus an mp4 of a timelapse of the past hour is posted. This has run fine for several months, but suddenly today I started receiving errors about the tweet length being over 140 characters:

`File "/usr/local/lib/python2.7/dist-packages/twitter/api.py", line 1008, in PostUpdate
    raise TwitterError("Text must be less than or equal to 140 characters.")
twitter.error.TwitterError: Text must be less than or equal to 140 characters.`

The text being posted was 125 characters, and the mp4 file attached was 1.9M. Both of these are very typical of what usually gets posted with no error, and there were no code changes between yesterday and today.

I tested posting the media with a modified tweet of 114 characters and it worked. So the problem seems to be with the 125 character text + media, though like I said this had worked up to today.

**Comment by jeremylow** [comment]
I'm not sure why it worked before, but doesn't now, but what I'm pretty sure is happening is this:

You've got a 125 character tweet. That tweet (without any URLs) counts for 125 characters. You add a `media` object onto that tweet, which gets uploaded and appended to your tweet. Currently, Twitter reserves 24 characters for a `media` object. (You can see this in https://github.com/bear/python-twitter/blob/master/testdata/get_help_configuration.json under `characters_reserved_per_media`). So what you end up with is 125 + 24 > 140, which is why your tweet gets bounced.

Do you happen to have any examples of tweets that previously worked or the account so I can check it out?

**Comment by tony-mcgee** [comment]
Thanks, that does make sense...here is an example of a 125 char tweet plus media posting on Mar 14: https://twitter.com/BuckroeWeather/status/841801244347883520

I started focusing on the text itself in detail and noticed the only difference was that one character that is usually a number was now a negative sign. Total length was the same (two digit positive value replaced with single digit neg value). Using the same media file, the following gives the 140 char limit error:
`Temp: 27.0°F || Dewpoint: -8.9°F || RH: 20% || Winds: WNW 15.2G22.8 mph || SLP: 1015.1 mb and rising || Past hour weathercam:`

and the following works:
`Temp: 27.0°F || Dewpoint: 18.9°F || RH: 20% || Winds: WNW 15.2G22.8 mph || SLP: 1015.1 mb and rising || Past hour weathercam:`

Only difference is the dewpoint being -8.9 in the failed tweet and 18.9 in the successful one. Can the dash (negative sign) somehow takes up more space than a numeric character? Yet another reason to complain about the cold weather on the US east coast!

**Comment by jeremylow** [comment]
No, it doesn't. I've tried it with the API and the web client and it shouldn't make a difference. My hunch is that there's some weirdness with the `attachement_url` param, but I have to look into it more.

**Comment by Radioguru** [comment]
I'm also seeing this issue....ironically, it's also while attempting to post weather station data.  I've just been working with it this week, so no prior experience with any of it.  I do not have anything in reference to media.  Mine is just a string of weather data.

I tried PostUpdates to see if it would handle it properly and it ended up splitting the same 105 character tweet, that PostUpdate said had to be less than 140 characters, at the 90 mark.

**Comment by jeremylow** [comment]
@Radioguru Can you link to the tweet or the string of data that you're trying to post?

**Comment by Radioguru** [comment]
Here is an example of a string of data (98) that gets the must be less that 140 error:
  "South Rim - Tmp:50.9F Wind:NNW at 0.0mph Bar:29.800 Rising Rapidly Rain:0.00in H/L Tmp:51.0F/50.9F"

Here are the links to a post that was split with PostUpdates:
https://twitter.com/GrandCanyonWx/status/870476608288702467
https://twitter.com/GrandCanyonWx/status/870475292732661760

Those had the "|" character and I have since dropped it for consistency.

Thanks for looking in to it!

**Comment by jeremylow** [comment]
The two strings below are being caught in the URL regex. I'll see if I can't figure out a fix for this. Thanks for the report.

```python

In [1]: import twitter

In [2]: [print(word, twitter.twitter_utils.is_url(word)) for word in "South Rim - Tmp:50.9F Wind:NNW at 0.0mph Bar:29.800 Rising Rapidly Rain:0.00in H/L Tmp:51.0F/50.9F".split(' ')]

"South" False
"Rim" False
"-" False
"Tmp:50.9F" True
"Wind:NNW" False
"at" False
"0.0mph" False
"Bar:29.800" True
"Rising" False
"Rapidly" False
"Rain:0.00in" True
"H/L" False
"Tmp:51.0F/50.9F" True
```

**Comment by Schwambo** [comment]
Has this been addressed? I am using the latest pypi version and are having the same issue:
A string generated from various words, symbols and line breaks gets posted every day. Usually the string is between 70-100 characterscharacter long.
Whenever there are “-“ included in the string the 140 character exception occurs

**Comment by jeremylow** [comment]
Fixed in release 3.4

```bash
In [8]: import twitter

In [9]: [print(word, twitter.twitter_utils.is_url(word)) th Rim - Tmp:50.9F Wind:NNW at 0.0mph Bar:29.800 Rising Rin H/L Tmp:51.0F/50.9F".split(' ')]
South False
Rim False
- False
Tmp:50.9F False
Wind:NNW False
at False
0.0mph False
Bar:29.800 False
Rising False
Rapidly False
Rain:0.00in False
H/L False
Tmp:51.0F/50.9F False

In [10]: twitter.twitter_utils.calc_expected_status_lengtmp:50.9F Wind:NNW at 0.0mph Bar:29.800 Rising Rapidly Rai:51.0F/50.9F")mp 
Out[10]: 98
```
